stage0/src/fw_cfg.rs

```diff
@@ -27,6 +30,23 @@ const FWCFG_PORT_DMA: u16 = 0x514;
 
 const SIGNATURE: &[u8] = b"QEMU";
 
+/// A single 4KiB buffer that is 4KiB page-aligned.
+#[repr(C, align(4096))]
+#[derive(Debug)]
+pub struct DmaBuffer {
+    data: [u8; Size4KiB::SIZE as usize],
+}
+
+static_assertions::assert_eq_size!(DmaBuffer, [u8; Size4KiB::SIZE as usize]);
+
+impl DmaBuffer {
```

Comment:
How about implementing Default for this struct?

Reply:
Done.